<?xml version="1.0"?>

<!-- map.launch -->
<launch>
    <!-- Waypoints -->
    <arg name="simulation" default="false"/>
    <include file="$(find igvc_navigation)/launch/waypoint_source.launch"
             unless="$(arg simulation)"/>

    <!-- Planner -->
    <node name="path_planner" pkg="igvc_navigation" type="d_lite_planner" output="screen" >
        <!-- Distance at which a node is considered a valid goal -->
        <param name="goal_range" type="double" value="0.3"/>
        <!-- Robot configuration space -->
        <param name="c_space" type="double" value="0.5"/>
        <!-- Path calculation rate -->
        <param name="rate" type="double" value="10"/>
        <!-- Max distance to a goal node -->
        <param name="maximum_distance" type="double" value="100.0"/>
        <!-- Publish a pointcloud of search nodes on the /expanded topic -->
        <param name="publish_expanded" type="bool" value="true"/>
        <!-- If, for some reason, the path planner decides no optimal path
        currently exists, use the previously generated path instead -->
        <param name="follow_old_path" type="bool" value="true"/>
        <!-- maximum occupancy probability of a cell before it is considered
        to have an infinite traversal cost -->
        <param name="occupancy_threshold" type="double" value="0.7"/>
    </node>
</launch>
